File: CPP/PSU.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"  // Mengimpor Komponen langsung

using namespace std;

class PSU : public Komponen {
private:
    string kapasitas;    // Kapasitas PSU (misalnya 500W, 650W)
    string efisiensi;    // Efisiensi PSU (misalnya "80+", "90+")
    string jenis_kabel;  // Jenis kabel PSU (misalnya "Modular", "Non-Modular")
public:
    // Constructor default
    PSU() : Komponen() {
        kapasitas = "";
        efisiensi = "";
        jenis_kabel = "";
    }

    // Constructor dengan parameter
    PSU(string merk, string nama, int harga, string kapasitas, string efisiensi, string jenis_kabel)
        : Komponen(merk, nama, harga) {  // Memanggil constructor dari Komponen
        this->kapasitas = kapasitas;
        this->efisiensi = efisiensi;
        this->jenis_kabel = jenis_kabel;
    }

    // Setter untuk kapasitas
    void setKapasitas(string kapasitas) {
        this->kapasitas = kapasitas;
    }

    // Setter untuk efisiensi
    void setEfisiensi(string efisiensi) {
        this->efisiensi = efisiensi;
    }

    // Setter untuk jenis_kabel
    void setJenisKabel(string jenis_kabel) {
        this->jenis_kabel = jenis_kabel;
    }

    // Getter untuk kapasitas
    string getKapasitas() {
        return kapasitas;
    }

    // Getter untuk efisiensi
    string getEfisiensi() {
        return efisiensi;
    }

    // Getter untuk jenis_kabel
    string getJenisKabel() {
        return jenis_kabel;
    }

    // Destructor
    ~PSU() {
        // Destructor kosong, karena tidak ada alokasi memori dinamis
    }
};

File: PYTHON/PSU.py
from Komponen import  Komponen

class PSU(Komponen):
    def __init__(self, merk='', nama='', harga=0, kapasitas='', efisiensi='', jenis_kabel=''):
        super().__init__(merk, nama, harga)  # Memanggil constructor dari Komponen
        self.kapasitas = kapasitas  # Kapasitas PSU (misalnya 500W, 650W)
        self.efisiensi = efisiensi  # Efisiensi PSU (misalnya "80+", "90+")
        self.jenis_kabel = jenis_kabel  # Jenis kabel PSU (misalnya "Modular", "Non-Modular")

    # Setter untuk kapasitas
    def set_kapasitas(self, kapasitas):
        self.kapasitas = kapasitas

    # Setter untuk efisiensi
    def set_efisiensi(self, efisiensi):
        self.efisiensi = efisiensi

    # Setter untuk jenis_kabel
    def set_jenis_kabel(self, jenis_kabel):
        self.jenis_kabel = jenis_kabel

    # Getter untuk kapasitas
    def get_kapasitas(self):
        return self.kapasitas

    # Getter untuk efisiensi
    def get_efisiensi(self):
        return self.efisiensi

    # Getter untuk jenis_kabel
    def get_jenis_kabel(self):
        return self.jenis_kabel
